internal/pkg/service/stream/mapping/table/column/renderer.go

```diff
@@ -86,3 +72,130 @@ func (r *Renderer) CSVValue(c Column, ctx recordctx.Context) (string, error) {
 
 	return "", errors.Errorf("unknown column type %T", c)
 }
+
+func (r *Renderer) mapPathCSVValue(c Path, ctx recordctx.Context) (string, error) {
+	bodyMap, err := ctx.BodyMap()
+	if err != nil {
+		return "", err
+	}
+
+	var value any
+
+	if c.Path == "" {
+		value = bodyMap
+	} else {
+		value, _, err = bodyMap.GetNested(c.Path)
+		if err != nil {
+			if c.DefaultValue != nil {
+				value = *c.DefaultValue
+			} else {
+				return "", errors.Wrapf(err, `path "%s" not found in the body`, c.Path)
+			}
+		}
+	}
+
+	if c.RawString {
+		if stringValue, ok := value.(string); ok {
+			return stringValue, nil
+		}
+	}
+
+	return json.EncodeString(value, false)
+}
+
+func (r *Renderer) jsonPathCSVValue(c Path, ctx recordctx.Context) (string, error) {
+	body, err := ctx.BodyBytes()
+	if err != nil {
+		return "", err
+	}
+
+	parser := r.fastjsonPool.Get()
+	defer r.fastjsonPool.Put(parser)
+
+	value, err := parser.ParseBytes(body)
+	if err != nil {
+		return "", err
+	}
+
+	path := orderedmap.PathFromStr(c.Path)
+
+	keys := []string{}
+	for _, step := range path {
+		var key string
+
+		switch step := step.(type) {
+		case orderedmap.MapStep:
+			key = step.Key()
+		case orderedmap.SliceStep:
+			key = strconv.Itoa(step.Index())
+		}
+
+		keys = append(keys, key)
+	}
+
+	var result any
+	var resultErr error
+
+	if len(keys) > 0 {
+		if value.Exists(keys...) {
+			value = value.Get(keys...)
+		} else {
+			resultErr = errors.New(fmt.Sprintf(`path "%s" not found in the body`, c.Path))
+		}
+	}
+
+	if resultErr == nil {
+		// Optimize TypeObject to avoid json decode and re-encode.
+		if value.Type() == fastjson.TypeObject {
+			return value.GetObject().String(), nil
+		}
+
+		result, resultErr = r.getJSONValue(value)
+	} else if c.DefaultValue != nil {
+		result = *c.DefaultValue
+		resultErr = nil
+	}
+
+	if resultErr != nil {
+		return "", resultErr
+	}
+
+	if c.RawString {
+		if stringValue, ok := result.(string); ok {
+			return stringValue, nil
+		}
+	}
+
+	return json.EncodeString(result, false)
+}
+
+func (r *Renderer) getJSONValue(value *fastjson.Value) (any, error) {
```

Comment:
this would be nice if it is inside library itself but did not find it. Consider moving it to utils. The `r` is not required by any code below.